consensus/clique/src/main/java/org/hyperledger/besu/consensus/clique/blockcreation/CliqueBlockScheduler.java

```diff
@@ -43,19 +44,24 @@ public class CliqueBlockScheduler extends DefaultBlockScheduler {
    * @param validatorProvider the validator provider
    * @param localNodeAddress the local node address
    * @param forksSchedule the transitions
+   * @param miningParameters the mining parameters to set the block period in seconds
    */
   public CliqueBlockScheduler(
       final Clock clock,
       final ValidatorProvider validatorProvider,
       final Address localNodeAddress,
-      final ForksSchedule<CliqueConfigOptions> forksSchedule) {
+      final ForksSchedule<CliqueConfigOptions> forksSchedule,
+      final MiningParameters miningParameters) {
     super(
-        parentHeader ->
-            (long)
-                forksSchedule
-                    .getFork(parentHeader.getNumber() + 1)
-                    .getValue()
-                    .getBlockPeriodSeconds(),
+        parentHeader -> {
+          final var blockPeriodSeconds =
+              forksSchedule
+                  .getFork(parentHeader.getNumber() + 1)
+                  .getValue()
+                  .getBlockPeriodSeconds();
+          miningParameters.setBlockPeriodSeconds(blockPeriodSeconds);
```

Comment:
Feels slightly odd to pass this into the constructor just to have this value set. 

I think I can see why you've done it though - the alternative of querying the forksSchedule at the point where we read from the miningParameters could be quite messy.

Reply:
Does seem like a bit hack putting it here to keep the MinerParameters synced with the current blockPeriodSeconds.

Rather than adding it here, I think we could add a block added listener in the CliqueBesuControllerBuilder to update the block period on the MiningParameters when we move to a new block height.

i.e. something like this in CliqueBesuControllerBuilder.createMiningCoordinator
```
protocolContext.getBlockchain().observeBlockAdded(o -> miningParameters.setBlockPeriodSeconds(forksSchedule.getFork(o.getBlock().getHeader().getNumber()).getValue().getBlockPeriodSeconds()));
```

Reply:
I was not happy with the solution too, so I have applied your suggestion, tnx